io/plugins/it.geosolutions.hale.io.appschema/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingGenerator.java

```diff
@@ -526,10 +527,48 @@ static QName tryInferNamespacePrefix(PropertyDefinition child) {
 						prefix = defaultType.getName().getPrefix();
 				}
 			}
+			else {
+				String newPrefix = getPrefix(dpd, namespaceURI);
+				if (StringUtils.isNotBlank(newPrefix)) {
+					prefix = newPrefix;
+				}
+			}
 		}
 		return new QName(namespaceURI, "", prefix);
 	}
 
+	/**
+	 * Tries to extract the namespace prefix from the provided
+	 * {@link PropertyDefinition} parameter, traversing over the nested complex
+	 * structure of XML definitions.
```

Comment:
Better JavaDoc would be:
```
Helper method that tries to find a good prefix for the provided namespace from the provided property definition, in practice it will traverse the XSD tree to find the namespace declaration. If no matching namespace declaration is found, NULL will be returned and the invoke will handle the situation, probably y generating a new namespace declaration and assigning and unique random prefix.
```